Tests/js/views/fields/assets/ez-relation-editview-tests.js

```diff
@@ -0,0 +1,419 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-relation-editview-tests', function (Y) {
+    var viewTest, registerTest, getFieldTest, getEmptyFieldTest;
+
+    viewTest = new Y.Test.Case({
+        name: "eZ Relation View test",
+
+        _getFieldDefinition: function (required) {
+            return {
+                isRequired: required
+            };
+        },
+
+        setUp: function () {
+            this.destinationContent = new Y.Mock();
+            this.destinationContentToJSON = {anythingJSONed: 'somethingJSONed'};
+            Y.Mock.expect(this.destinationContent, {
+                method: 'toJSON',
+                returns: this.destinationContentToJSON
+            });
+            this.fieldDefinitionIdentifier= "niceField";
+            this.fieldDefinition = {
+                fieldType: "ezobjectrelation",
+                identifier: this.fieldDefinitionIdentifier,
+                isRequired: false
+            };
+            this.field = {fieldValue: {destinationContentId: 45}};
+            this.jsonContent = {};
+            this.jsonContentType = {};
+            this.jsonVersion = {};
+            this.content = new Y.Mock();
+            this.version = new Y.Mock();
+            this.contentType = new Y.Mock();
+            Y.Mock.expect(this.content, {
+                method: 'toJSON',
+                returns: this.jsonContent
+            });
+            Y.Mock.expect(this.version, {
+                method: 'toJSON',
+                returns: this.jsonVersion
+            });
+            Y.Mock.expect(this.contentType, {
+                method: 'toJSON',
+                returns: this.jsonContentType
+            });
+
+            this.view = new Y.eZ.RelationEditView({
+                container: '.container',
+                field: this.field,
+                fieldDefinition: this.fieldDefinition,
+                content: this.content,
+                version: this.version,
+                contentType: this.contentType,
+                destinationContent: this.destinationContent,
+                destinationContentId: null,
+            });
+        },
+
+        tearDown: function () {
+            this.view.destroy();
+        },
+
+        _testAvailableVariables: function (required, expectRequired) {
+            var fieldDefinition = this._getFieldDefinition(required),
+                that = this;
+
+            this.view.set('fieldDefinition', fieldDefinition);
+
+            this.view.template = function (variables) {
+                Y.Assert.isObject(variables, "The template should receive some variables");
+                Y.Assert.areEqual(9, Y.Object.keys(variables).length, "The template should receive 9 variables");
```

Comment:
there are 9 nine variables in the templates but you are testing only 6. `destinationContent`, `loadingError` and `isEmpty` are not tested anywhere AFAICS.